server/src/main/java/org/elasticsearch/cluster/metadata/IndexAbstractionResolver.java

```diff
@@ -123,11 +124,18 @@ private void resolveIndexAbstraction(
         final boolean includeDataStreams,
         final Set<String> remoteExpressions
     ) {
+        if (localIndexExpression.isEmpty()) {
+            throw invalidExpression(false);
```

Comment:
Nit: I'd add `invalidExclusion()` and `invalidExpression()` methods to make the call more self-explanatory (as opposed to the raw boolean arg)